packages/@webex/plugin-cc/src/services/Task/index.ts

```diff
@@ -0,0 +1,93 @@
+import EventEmitter from 'events';
+import {ICall, LINE_EVENTS, LocalMicrophoneStream, createMicrophoneStream} from '@webex/calling';
+import Services from '..';
+import {getErrorDetails} from '../core/Utils';
+import WebCallingService from '../WebCallingService';
+import {LoginOption} from '../../types';
+import {CC_FILE} from '../../constants';
+import {CC_EVENTS} from '../config/types';
+
+export default class Task extends EventEmitter {
+  private localAudioStream: LocalMicrophoneStream;
+  private services: Services; // This will be used later for invoking Contact APIs
+  private webCallingService: WebCallingService;
+  private taskPayload: any;
+  private call: ICall;
+
+  constructor(services: Services, webCallingService: WebCallingService) {
+    super();
+    this.services = services;
+    this.webCallingService = webCallingService;
+    this.registerTaskListeners();
+    this.registerCallingEvent();
+  }
+
+  private registerCallingEvent() {
+    this.webCallingService.on(LINE_EVENTS.INCOMING_CALL, (call) => {
+      if (this.taskPayload) {
+        this.emit('task:incoming', this.taskPayload);
+      } else {
+        this.call = call;
+      }
+    });
+  }
+
+  private registerTaskListeners() {
+    // TODO: This event reception approach will be changed once state machine is implemented
+    this.services.webSocketManager.on('message', (event) => {
+      const payload = JSON.parse(event);
+      switch (payload.data.type) {
+        case CC_EVENTS.AGENT_CONTACT_RESERVED:
+          this.taskPayload = payload.data;
+          if (this.webCallingService.loginOption !== LoginOption.BROWSER) {
+            this.emit('task:incoming', payload.data);
+          } else if (this.call) {
```

Comment:
Just curious, is it possible to encounter a situation where we do not have the calling service created and we are receicing web socket events (for BROWSER) - I would assume if either was present we would be having both (as per the initialisation we do in cc.ts).
Only reason I'm asking is because we can avoid this double if condition. Just a query, however.